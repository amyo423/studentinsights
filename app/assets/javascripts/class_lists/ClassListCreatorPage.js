import React from 'react';
import _ from 'lodash';
import {sortByGrade} from '../helpers/SortHelpers';
import {
  fetchGradeLevelsJson,
  fetchStudentsJson,
  fetchClassListJson,
  postClassList
} from './api';
import Loading from '../components/Loading';
import {initialStudentIdsByRoom} from './studentIdsByRoomFunctions';
import ClassListCreatorWorkflow from './ClassListCreatorWorkflow';
import uuidv4 from 'uuid/v4';

export const STEPS = [
  'Choose your grade',
  'Make a plan',
  'Create your classrooms',
  'Notes to principal',
  'Submit to principal'
];


// Root page component.
// This component manages state transitions and hands off requests to the server
// and rendering to other components.  On state changes, it saves to the server
// with some throttling to prevent too much server communication.
export default class ClassListCreatorPage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      stepIndex: 0,
      isEditable: true,

      // tracking in-flight server requests
      hasFetchedStudents: false,
      hasFetchedGradeLevels: false,
      hasFetchedClassList: false,

      // choosing school and grade
      schools: null, // from server
      gradeLevelsNextYear: null, // from server
      schoolId: null,
      gradeLevelNextYear: null,

      // workspace
      educators: null, // from server
      students: null, // from server
      classroomsCount: 4,
      authors: [],
      planText: '',
      studentIdsByRoom: null,
      principalNoteText: ''
    };

    this.doSaveChanges = _.throttle(this.doSaveChanges, 5000);
    this.fetchGradeLevels = this.fetchGradeLevels.bind(this);
    this.fetchStudents = this.fetchStudents.bind(this);
    this.fetchClassList = this.fetchClassList.bind(this);
    this.onFetchedGradeLevels = this.onFetchedGradeLevels.bind(this);
    this.onFetchedStudents = this.onFetchedStudents.bind(this);
    this.onFetchedClassList = this.onFetchedClassList.bind(this);
    this.onStepChanged = this.onStepChanged.bind(this);
    this.onSchoolIdChanged = this.onSchoolIdChanged.bind(this);
    this.onGradeLevelNextYearChanged = this.onGradeLevelNextYearChanged.bind(this);
    this.onClassroomsCountIncremented = this.onClassroomsCountIncremented.bind(this);
    this.onPlanTextChanged = this.onPlanTextChanged.bind(this);
    this.onAuthorsChanged = this.onAuthorsChanged.bind(this);
    this.onClassListsChanged = this.onClassListsChanged.bind(this);
    this.onPrincipalNoteChanged = this.onPrincipalNoteChanged.bind(this);
  }

  componentDidMount() {
    this.doSizePage();
    window.addEventListener('beforeunload', this.onBeforeUnload);
    this.triggerEffects();
    this.installDebugHook();
  }

  componentDidUpdate() {
    this.doSaveChanges();
    this.triggerEffects();
  }

  componentWillUnmount() {
    if (this.doSaveChanges.flush) this.doSaveChanges.flush(); // flush any queued changes
    window.removeEventListener('beforeunload', this.onBeforeUnload);
  }

  // Describes which steps are available to be navigated to,
  // not which data has been loaded for.  Steps handle showing their own loading
  // states based on data.
  availableSteps() {
    const {
      schoolId,
      gradeLevelNextYear
    } = this.state;

    return (schoolId === null || gradeLevelNextYear === null)
      ? [0]
      : [0, 1, 2, 3, 4];
  }

  // This is a debug hook for iterating on particular production data sets locally
  // during development.
  installDebugHook() {
    window.forceDebug = this.onForceDebug.bind(this);
  }

  doSizePage() {
    const {disableSizing} = this.props;
    if (disableSizing) return;
    
    // Reach outside component to change styles for page and conatiner, to take up
    // the entire vertical height.
    window.document.documentElement.style.height = '100%';
    window.document.body.style.height = '100%';
    window.document.body.style.display = 'flex';
    window.document.body.style['flex-direction'] = 'column';
    window.document.getElementById('main').style.flex = 1;
    window.document.getElementById('main').style.display = 'flex';
    
    // Prevent horizontal scrollbar from showing.
    window.document.body.style['min-width'] = '1000px';
  }
  
  // Trigger fetches and other initialization
  triggerEffects() {
    const {defaultWorkspaceId} = this.props;
    const {
      workspaceId,
      stepIndex,
      schoolId,
      gradeLevelNextYear,
      schools,
      gradeLevelsNextYear,
      students,
      educators,
      classroomsCount,
      studentIdsByRoom
    } = this.state;
    
    // Update URL (no state change)
    if (workspaceId && window.location.pathname !== `/classlists/${workspaceId}`) {
      this.doReplaceState();
    }

    // Assign workspaceId
    if (!workspaceId) {
      return this.setState({workspaceId: defaultWorkspaceId || uuidv4()});
    }

    // New classlist
    // () => {schools, grades}
    if (schools === null || gradeLevelsNextYear === null) {
      return this.fetchGradeLevels();
    }

    // The user has chosen a school and grade and moved past the first screen.
    // Fetch the students for that grade, and the list of educators.
    // (schoolId, gradeLevelNextYear) => {students, educators}
    if (schoolId !== null && gradeLevelNextYear !== null && (students == null || educators == null) && (stepIndex !== 0 || defaultWorkspaceId)) {
      return this.fetchStudents();
    }

    // If we're navigating to `CreateYourLists` for the first time and
    // don't have classroom lists yet, create the default
    if (stepIndex == 2 && students !== null && studentIdsByRoom === null) {
      const updatedStudentIdsByRoom = initialStudentIdsByRoom(classroomsCount, students);
      return this.setState({studentIdsByRoom: updatedStudentIdsByRoom});
    }
     
    // Loading previous session
    if (workspaceId && defaultWorkspaceId && studentIdsByRoom === null) {
      return this.fetchClassList();
    }
  }

  fetchGradeLevels() {
    const {hasFetchedGradeLevels, workspaceId} = this.state;
    if (hasFetchedGradeLevels) return;
    this.setState({hasFetchedGradeLevels: true});
    return fetchGradeLevelsJson(workspaceId).then(this.onFetchedGradeLevels);
  }

  fetchStudents() {
    const {hasFetchedStudents, workspaceId, gradeLevelNextYear, schoolId} = this.state;
    if (hasFetchedStudents) return;
    this.setState({hasFetchedStudents: true});
    return fetchStudentsJson({workspaceId, gradeLevelNextYear, schoolId}).then(this.onFetchedStudents);
  }

  doReplaceState() {
    const {disableHistory} = this.props;
    const {workspaceId} = this.state;
    const path = `/classlists/${workspaceId}`;
    if (!disableHistory) {
      window.history.replaceState({}, null, path);
    }
  }

  // This method is throttled.
  doSaveChanges() {
    const {
      isEditable,
      workspaceId,
      stepIndex,
      schoolId,
      gradeLevelNextYear,
      authors,
      classroomsCount,
      planText,
      studentIdsByRoom,
      principalNoteText
    } = this.state;

    // View-only
    if (!isEditable) return;

    // Don't save until they choose a grade level and school
    if (!workspaceId || stepIndex === 0 || !schoolId || !gradeLevelNextYear) return;
    
    const payload = {
      workspaceId,
      stepIndex,
      schoolId,
      gradeLevelNextYear,
      authors,
      classroomsCount,
      planText,
      studentIdsByRoom,
      principalNoteText,
      clientNowMs: moment.utc().unix()
    };
    postClassList(payload);
  }

  fetchClassList() {
    const {hasFetchedClassList, workspaceId} = this.state;
    if (hasFetchedClassList) return;
    this.setState({hasFetchedClassList: true});
    return fetchClassListJson(workspaceId).then(this.onFetchedClassList);
  }

  // TODO(kr) check this on IE
  onBeforeUnload(event) {
    const isDirty = true;
    return (isDirty)
      ? 'You have unsaved changes.'
      : undefined;
  }

  onForceDebug(nextState) {
    const {gradeLevelNextYear, students, studentIdsByRoom} = nextState;
    this.setState({gradeLevelNextYear, students, studentIdsByRoom, stepIndex: 2});
  }

  onFetchedGradeLevels(json) {
    const gradeLevelsNextYear = json.grade_levels_next_year.sort(sortByGrade);
    const {schools} = json;

    this.setState({
      schools,
      gradeLevelsNextYear
    });
  }

  onFetchedStudents(json) {
    const {isEditable, authors} = this.state;
    const {educators, students} = json;

    // initialize if we just got the list of educators, and there's no one set yet
    const updatedAuthors = (authors.length === 0 && isEditable)
      ? educators.filter(educator => educator.id === json.current_educator_id)
      : authors;
    this.setState({
      students,
      educators,
      authors: updatedAuthors
    });
  }

  onFetchedClassList(responseJson) {
    const isEditable = responseJson.is_editable;
    const classList = responseJson.class_list;
    const workspaceId = classList.workspace_id;
    const schoolId = classList.school_id;
    const gradeLevelNextYear = classList.grade_level_next_year;
    const {
      authors,
      classroomsCount,
      planText,
      studentIdsByRoom,
      principalNoteText
    } = classList.json;
    this.setState({
      workspaceId,
      isEditable,
      schoolId,
      gradeLevelNextYear,
      authors,
      classroomsCount,
      planText,
      studentIdsByRoom,
      principalNoteText,
      stepIndex: 0, // Ignore last `stepIndex`
    });
  }

  onStepChanged(stepIndex) {
    this.setState({stepIndex});
  }

  // TODO(kr) warn about resetting students?
  onSchoolIdChanged(schoolId) {
    this.setState({schoolId});
  }

  // TODO(kr) warn about resetting students?
  onGradeLevelNextYearChanged(gradeLevelNextYear) {
    this.setState({gradeLevelNextYear});
  }

  onAuthorsChanged(authors) {
    this.setState({authors}); 
  }

  // TODO(kr) warn about resetting students?
  onClassroomsCountIncremented(delta) {
    const {classroomsCount} = this.state;
    const updatedClassroomsCount = classroomsCount + delta;
    this.setState({classroomsCount: updatedClassroomsCount});
  }

  onPlanTextChanged(planText) {
    this.setState({planText});
  }

  onClassListsChanged(studentIdsByRoom) {
    this.setState({studentIdsByRoom});
  }

  onPrincipalNoteChanged(principalNoteText) {
    this.setState({principalNoteText});
  }

  render() {
    const {workspaceId} = this.state;
    if (!workspaceId) return <Loading />;

    const availableSteps = this.availableSteps();
    return (
      <ClassListCreatorWorkflow
        {...this.state}
        steps={STEPS}
        availableSteps={availableSteps}
        onStepChanged={this.onStepChanged}
        onSchoolIdChanged={this.onSchoolIdChanged}
        onGradeLevelNextYearChanged={this.onGradeLevelNextYearChanged}
        onEducatorsChanged={this.onAuthorsChanged}
        onClassroomsCountIncremented={this.onClassroomsCountIncremented}
        onPlanTextChanged={this.onPlanTextChanged}
        onClassListsChanged={this.onClassListsChanged}
        onPrincipalNoteChanged={this.onPrincipalNoteChanged}
      />
    );
  }
}
ClassListCreatorPage.propTypes = {
  defaultWorkspaceId: React.PropTypes.string,
  disableHistory: React.PropTypes.bool,
  disableSizing: React.PropTypes.bool
};


