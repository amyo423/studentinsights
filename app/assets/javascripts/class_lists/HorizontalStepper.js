import React from 'react';
import Circle from '../components/Circle';
import Button from '../components/Button';

// Shows progress through a set of steps for the classroom creator process.
// This also controls the navigation buttons
export default class HorizontalStepper extends React.Component {
  constructor(props) {
    super(props);

    this.onStepChanged = this.onStepChanged.bind(this);
    this.onPreviousClicked = this.onPreviousClicked.bind(this);
    this.onNextClicked = this.onNextClicked.bind(this);
  }

  onStepChanged(stepIndex) {
    const {onStepChanged, availableSteps} = this.props;
    if (availableSteps.indexOf(stepIndex) === -1) return;
    onStepChanged(stepIndex);
  }

  onPreviousClicked() {
    const {stepIndex} = this.props;
    this.onStepChanged(stepIndex - 1);
  }

  onNextClicked() {
    const {stepIndex} = this.props;
    this.onStepChanged(stepIndex + 1);
  }

  render() {
    const {steps, availableSteps, isEditable, renderFn, style, contentStyle} = this.props;
    const currentStepIndex = this.props.stepIndex;
    return (
      <div className="HorizontalStepper" style={{...styles.root, ...style}}>
        <div style={styles.banner}>
          {steps.map((step, stepIndex) => {
            return (
              <span
                key={stepIndex}
                style={this.renderStyleForBannerItem(stepIndex, currentStepIndex, availableSteps)}
                onClick={this.onStepChanged.bind(this, stepIndex)}>
                <Circle
                  text={`${stepIndex+1}`}
                  color={this.renderColorForCircle(stepIndex, availableSteps)}
                  style={{verticalAlign: 'middle'}}/>
                <span style={styles.bannerText}>{step}</span>
              </span>
            );
          })}
          {isEditable && <div>readonly</div>}
        </div>
        <div style={{...styles.content, ...contentStyle}}>
          {renderFn(currentStepIndex, steps[currentStepIndex])}
        </div>
        {this.renderNavigationButtons()}
      </div>
    );
  }

  renderColorForCircle(index, availableSteps) {
    return (availableSteps.indexOf(index) !== -1)
      ? '#3177c9'
      : '#ccc';
  }

  renderStyleForBannerItem(index, currentIndex, availableSteps) {
    if (index === currentIndex) {
      return {
        ...styles.bannerItem,
        borderColor: '#3177c9',
        backgroundColor: 'rgba(49, 119, 201, 0.25)'
      };
    }

    return (availableSteps.indexOf(index) !== -1)
      ? styles.bannerItem
      : {...styles.bannerItem, color: '#ccc' };
  }

  renderNavigationButtons() {
    const {stepIndex, availableSteps} = this.props;
    const shouldShowNext = (availableSteps.indexOf(stepIndex + 1) !== -1);
    const shouldShowPrevious = (availableSteps.indexOf(stepIndex - 1) !== -1);
    return (
      <div style={styles.buttonStrip}>
        {shouldShowPrevious
          ? <Button style={{margin: 10}} onClick={this.onPreviousClicked}>{`< Back `}</Button>
          : <div />}
        {shouldShowNext
          ? <Button style={{margin: 10}} onClick={this.onNextClicked}>Next ></Button>
          : <div />}
      </div>
    );
  }
}
HorizontalStepper.propTypes = {
  steps: React.PropTypes.arrayOf(React.PropTypes.string).isRequired,
  availableSteps: React.PropTypes.arrayOf(React.PropTypes.number).isRequired,
  stepIndex: React.PropTypes.number.isRequired,
  onStepChanged: React.PropTypes.func.isRequired,
  isEditable: React.PropTypes.booel.isRequired,
  renderFn: React.PropTypes.func.isRequired,
  style: React.PropTypes.object,
  contentStyle: React.PropTypes.object
};

const styles = {
  root: {
    paddingTop: 15
  },
  content: {
    borderTop: '1px solid #ccc',
    marginTop: 10
  },
  banner:{
    fontSize: 12,
    paddingLeft: 15
  },
  bannerItem: {
    padding: 8,
    paddingRight: 10,
    marginLeft: 5,
    marginRight: 10,
    cursor: 'pointer',
    border: '1px solid white',
    borderRadius: 3,
    borderColor: 'white'
  },
  bannerText: {
    paddingLeft: 5
  },
  check: {},
  buttonStrip: {
    display: 'flex',
    justifyContent: 'space-between',
    margin: 10
  },
  divider: {
    width: '2em',
    height: 1,
    borderTop: '1px solid #eee'
  }
};
