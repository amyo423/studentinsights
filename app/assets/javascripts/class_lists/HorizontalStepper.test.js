import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import HorizontalStepper from './HorizontalStepper';
import {STEPS} from './ClassListCreatorPage';

export function testProps(props = {}) {
  return {
    availableSteps: STEPS.map((step, index) => index),
    steps: STEPS,
    stepIndex: 2,
    isEditable: true,
    isDirty: false,
    onStepChanged: jest.fn(),
    renderFn(stepIndex, step) {
      return <div style={{border: '1px solid #eee', padding: 20}}>contents for {step}</div>;
    },
    ...props
  };
}

it('renders without crashing', () => {
  const el = document.createElement('div');
  const props = testProps();
  ReactDOM.render(<HorizontalStepper {...props} />, el);
});

it('snapshots', () => {
  const props = testProps();
  const tree = renderer
    .create(<HorizontalStepper {...props} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('snapshots when readonly', () => {
  const props = testProps({ isEditable: false });
  const tree = renderer
    .create(<HorizontalStepper {...props} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});