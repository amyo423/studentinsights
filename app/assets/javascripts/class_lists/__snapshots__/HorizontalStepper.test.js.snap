// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<div
  className="HorizontalStepper"
  style={
    Object {
      "paddingTop": 15,
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "fontSize": 12,
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "paddingLeft": 15,
        }
      }
    >
      <span
        onClick={[Function]}
        style={
          Object {
            "border": "1px solid white",
            "borderColor": "white",
            "borderRadius": 3,
            "cursor": "pointer",
            "marginLeft": 5,
            "marginRight": 10,
            "padding": 8,
            "paddingRight": 10,
          }
        }
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg viewBox=\\"0 0 24 24\\" style=\\"display: inline-block; color: #3177c9; fill: #3177c9; height: 24px; width: 24px; user-select: none; transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms; font-size: 24px;\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"10\\"></circle><text x=\\"12\\" y=\\"16\\" text-anchor=\\"middle\\" font-size=\\"12\\" fill=\\"#fff\\">1</text></svg>",
            }
          }
          style={
            Object {
              "display": "inline-block",
              "verticalAlign": "middle",
            }
          }
        />
        <span
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          Choose your grade
        </span>
      </span>
      <span
        onClick={[Function]}
        style={
          Object {
            "border": "1px solid white",
            "borderColor": "white",
            "borderRadius": 3,
            "cursor": "pointer",
            "marginLeft": 5,
            "marginRight": 10,
            "padding": 8,
            "paddingRight": 10,
          }
        }
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg viewBox=\\"0 0 24 24\\" style=\\"display: inline-block; color: #3177c9; fill: #3177c9; height: 24px; width: 24px; user-select: none; transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms; font-size: 24px;\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"10\\"></circle><text x=\\"12\\" y=\\"16\\" text-anchor=\\"middle\\" font-size=\\"12\\" fill=\\"#fff\\">2</text></svg>",
            }
          }
          style={
            Object {
              "display": "inline-block",
              "verticalAlign": "middle",
            }
          }
        />
        <span
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          Make a plan
        </span>
      </span>
      <span
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(49, 119, 201, 0.25)",
            "border": "1px solid white",
            "borderColor": "#3177c9",
            "borderRadius": 3,
            "cursor": "pointer",
            "marginLeft": 5,
            "marginRight": 10,
            "padding": 8,
            "paddingRight": 10,
          }
        }
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg viewBox=\\"0 0 24 24\\" style=\\"display: inline-block; color: #3177c9; fill: #3177c9; height: 24px; width: 24px; user-select: none; transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms; font-size: 24px;\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"10\\"></circle><text x=\\"12\\" y=\\"16\\" text-anchor=\\"middle\\" font-size=\\"12\\" fill=\\"#fff\\">3</text></svg>",
            }
          }
          style={
            Object {
              "display": "inline-block",
              "verticalAlign": "middle",
            }
          }
        />
        <span
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          Create your classrooms
        </span>
      </span>
      <span
        onClick={[Function]}
        style={
          Object {
            "border": "1px solid white",
            "borderColor": "white",
            "borderRadius": 3,
            "cursor": "pointer",
            "marginLeft": 5,
            "marginRight": 10,
            "padding": 8,
            "paddingRight": 10,
          }
        }
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg viewBox=\\"0 0 24 24\\" style=\\"display: inline-block; color: #3177c9; fill: #3177c9; height: 24px; width: 24px; user-select: none; transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms; font-size: 24px;\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"10\\"></circle><text x=\\"12\\" y=\\"16\\" text-anchor=\\"middle\\" font-size=\\"12\\" fill=\\"#fff\\">4</text></svg>",
            }
          }
          style={
            Object {
              "display": "inline-block",
              "verticalAlign": "middle",
            }
          }
        />
        <span
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          Notes to principal
        </span>
      </span>
      <span
        onClick={[Function]}
        style={
          Object {
            "border": "1px solid white",
            "borderColor": "white",
            "borderRadius": 3,
            "cursor": "pointer",
            "marginLeft": 5,
            "marginRight": 10,
            "padding": 8,
            "paddingRight": 10,
          }
        }
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg viewBox=\\"0 0 24 24\\" style=\\"display: inline-block; color: #3177c9; fill: #3177c9; height: 24px; width: 24px; user-select: none; transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms; font-size: 24px;\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"10\\"></circle><text x=\\"12\\" y=\\"16\\" text-anchor=\\"middle\\" font-size=\\"12\\" fill=\\"#fff\\">5</text></svg>",
            }
          }
          style={
            Object {
              "display": "inline-block",
              "verticalAlign": "middle",
            }
          }
        />
        <span
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          Submit to principal
        </span>
      </span>
    </div>
  </div>
  <div
    style={
      Object {
        "borderTop": "1px solid #ccc",
        "marginTop": 10,
      }
    }
  >
    <div
      style={
        Object {
          "border": "1px solid #eee",
          "padding": 20,
        }
      }
    >
      contents for 
      Create your classrooms
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "margin": 10,
      }
    }
  >
    <div
      className="Hover"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={undefined}
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#4A90E2",
            "border": "none",
            "borderRadius": 28,
            "color": "white",
            "cursor": "pointer",
            "fontSize": 14,
            "margin": 10,
            "padding": "8px 25px",
            "textDecoration": "none",
          }
        }
      >
        &lt; Back 
      </button>
    </div>
    <div
      className="Hover"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={undefined}
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#4A90E2",
            "border": "none",
            "borderRadius": 28,
            "color": "white",
            "cursor": "pointer",
            "fontSize": 14,
            "margin": 10,
            "padding": "8px 25px",
            "textDecoration": "none",
          }
        }
      >
        Next &gt;
      </button>
    </div>
  </div>
</div>
`;

exports[`snapshots when readonly 1`] = `
<div
  className="HorizontalStepper"
  style={
    Object {
      "paddingTop": 15,
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "fontSize": 12,
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "paddingLeft": 15,
        }
      }
    >
      <span
        onClick={[Function]}
        style={
          Object {
            "border": "1px solid white",
            "borderColor": "white",
            "borderRadius": 3,
            "cursor": "pointer",
            "marginLeft": 5,
            "marginRight": 10,
            "padding": 8,
            "paddingRight": 10,
          }
        }
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg viewBox=\\"0 0 24 24\\" style=\\"display: inline-block; color: #3177c9; fill: #3177c9; height: 24px; width: 24px; user-select: none; transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms; font-size: 24px;\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"10\\"></circle><text x=\\"12\\" y=\\"16\\" text-anchor=\\"middle\\" font-size=\\"12\\" fill=\\"#fff\\">1</text></svg>",
            }
          }
          style={
            Object {
              "display": "inline-block",
              "verticalAlign": "middle",
            }
          }
        />
        <span
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          Choose your grade
        </span>
      </span>
      <span
        onClick={[Function]}
        style={
          Object {
            "border": "1px solid white",
            "borderColor": "white",
            "borderRadius": 3,
            "cursor": "pointer",
            "marginLeft": 5,
            "marginRight": 10,
            "padding": 8,
            "paddingRight": 10,
          }
        }
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg viewBox=\\"0 0 24 24\\" style=\\"display: inline-block; color: #3177c9; fill: #3177c9; height: 24px; width: 24px; user-select: none; transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms; font-size: 24px;\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"10\\"></circle><text x=\\"12\\" y=\\"16\\" text-anchor=\\"middle\\" font-size=\\"12\\" fill=\\"#fff\\">2</text></svg>",
            }
          }
          style={
            Object {
              "display": "inline-block",
              "verticalAlign": "middle",
            }
          }
        />
        <span
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          Make a plan
        </span>
      </span>
      <span
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(49, 119, 201, 0.25)",
            "border": "1px solid white",
            "borderColor": "#3177c9",
            "borderRadius": 3,
            "cursor": "pointer",
            "marginLeft": 5,
            "marginRight": 10,
            "padding": 8,
            "paddingRight": 10,
          }
        }
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg viewBox=\\"0 0 24 24\\" style=\\"display: inline-block; color: #3177c9; fill: #3177c9; height: 24px; width: 24px; user-select: none; transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms; font-size: 24px;\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"10\\"></circle><text x=\\"12\\" y=\\"16\\" text-anchor=\\"middle\\" font-size=\\"12\\" fill=\\"#fff\\">3</text></svg>",
            }
          }
          style={
            Object {
              "display": "inline-block",
              "verticalAlign": "middle",
            }
          }
        />
        <span
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          Create your classrooms
        </span>
      </span>
      <span
        onClick={[Function]}
        style={
          Object {
            "border": "1px solid white",
            "borderColor": "white",
            "borderRadius": 3,
            "cursor": "pointer",
            "marginLeft": 5,
            "marginRight": 10,
            "padding": 8,
            "paddingRight": 10,
          }
        }
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg viewBox=\\"0 0 24 24\\" style=\\"display: inline-block; color: #3177c9; fill: #3177c9; height: 24px; width: 24px; user-select: none; transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms; font-size: 24px;\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"10\\"></circle><text x=\\"12\\" y=\\"16\\" text-anchor=\\"middle\\" font-size=\\"12\\" fill=\\"#fff\\">4</text></svg>",
            }
          }
          style={
            Object {
              "display": "inline-block",
              "verticalAlign": "middle",
            }
          }
        />
        <span
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          Notes to principal
        </span>
      </span>
      <span
        onClick={[Function]}
        style={
          Object {
            "border": "1px solid white",
            "borderColor": "white",
            "borderRadius": 3,
            "cursor": "pointer",
            "marginLeft": 5,
            "marginRight": 10,
            "padding": 8,
            "paddingRight": 10,
          }
        }
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<svg viewBox=\\"0 0 24 24\\" style=\\"display: inline-block; color: #3177c9; fill: #3177c9; height: 24px; width: 24px; user-select: none; transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms; font-size: 24px;\\"><circle cx=\\"12\\" cy=\\"12\\" r=\\"10\\"></circle><text x=\\"12\\" y=\\"16\\" text-anchor=\\"middle\\" font-size=\\"12\\" fill=\\"#fff\\">5</text></svg>",
            }
          }
          style={
            Object {
              "display": "inline-block",
              "verticalAlign": "middle",
            }
          }
        />
        <span
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          Submit to principal
        </span>
      </span>
    </div>
    <div
      style={
        Object {
          "background": "#666",
          "borderRadius": 3,
          "color": "#eee",
          "display": "inline-block",
          "marginRight": 20,
          "padding": 6,
          "paddingLeft": 12,
          "paddingRight": 12,
        }
      }
    >
      readonly
    </div>
  </div>
  <div
    style={
      Object {
        "borderTop": "1px solid #ccc",
        "marginTop": 10,
      }
    }
  >
    <div
      style={
        Object {
          "border": "1px solid #eee",
          "padding": 20,
        }
      }
    >
      contents for 
      Create your classrooms
    </div>
  </div>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
        "margin": 10,
      }
    }
  >
    <div
      className="Hover"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={undefined}
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#4A90E2",
            "border": "none",
            "borderRadius": 28,
            "color": "white",
            "cursor": "pointer",
            "fontSize": 14,
            "margin": 10,
            "padding": "8px 25px",
            "textDecoration": "none",
          }
        }
      >
        &lt; Back 
      </button>
    </div>
    <div
      className="Hover"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={undefined}
    >
      <button
        onClick={[Function]}
        style={
          Object {
            "background": "#4A90E2",
            "border": "none",
            "borderRadius": 28,
            "color": "white",
            "cursor": "pointer",
            "fontSize": 14,
            "margin": 10,
            "padding": "8px 25px",
            "textDecoration": "none",
          }
        }
      >
        Next &gt;
      </button>
    </div>
  </div>
</div>
`;
